internal/workflow.go

```diff
@@ -1530,10 +1530,26 @@ func WithLocalActivityOptions(ctx Context, options LocalActivityOptions) Context
 
 	opts.ScheduleToCloseTimeout = options.ScheduleToCloseTimeout
 	opts.StartToCloseTimeout = options.StartToCloseTimeout
-	opts.RetryPolicy = options.RetryPolicy
+	opts.RetryPolicy = applyRetryPolicyDefaultsForLocalActivity(options.RetryPolicy, opts.ScheduleToCloseTimeout)
 	return ctx1
 }
 
+func applyRetryPolicyDefaultsForLocalActivity(policy *RetryPolicy, timeout time.Duration) *RetryPolicy {
+	if policy == nil {
+		policy = &RetryPolicy{}
+	}
+	if policy.BackoffCoefficient == 0 {
+		policy.BackoffCoefficient = 2
+	}
+	if policy.InitialInterval == 0 {
+		policy.InitialInterval = 50 * time.Millisecond
+	}
+	if policy.MaximumInterval == 0 {
+		policy.MaximumInterval = timeout
```

Comment:
I think this is not what you intended?

Reply:
Why? Maximum interval should be equal to schedule to close timeout. That's what our [documentation](https://github.com/temporalio/sdk-go/blob/0a5c65d55f693ea569b5ca7009eed9d4ea9a5f64/internal/activity.go#L148) was saying, which wasn't true. So this line is fixing it.

Reply:
Yes, you are right, we should follow our [documentation](https://github.com/temporalio/sdk-go/blob/e8455cbe8014d2587a2b72ce153ae490160d9b11/internal/client.go#L556) and default max interval to 100x initial interval.